### PR TITLE
Login: fix crash when dismissing Google view

### PR DESCRIPTION
**Pull request by nheagy**
refs #7675

To test:
- Tap Log in with Google button
- Tap Cancel in the upper left
- _App should not crash_

Needs review: @aerych 

*Note*: this was branched from #8010 and will include its changes until merged.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift [hunk 1]
```diff
@@ -451,9 +451,10 @@ extension LoginEmailViewController {
 }
 
 extension LoginEmailViewController: GIDSignInDelegate {
-    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error!) {
+    func sign(_ signIn: GIDSignIn?, didSignInFor user: GIDGoogleUser?, withError error: Error?) {
         // TODO: finish implementing wpcom login via Google code
-        guard let token = user.authentication.idToken,
+        guard let user = user,
+            let token = user.authentication.idToken,
                 let email = user.profile.email else {
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift]: Nitpicky on style but maybe we could align the `let` statements?

**Comment by aerych** [comment]
Howdy Nate!
Just one super nitpicky comment. 
Took it for a spin in the simulator and the crash is resolved :)  
:shipit: when its time.

**Comment by nheagy** [comment]
@aerych I plan to merge this in the morning, unless you see something else super bad 😜

**Comment by nheagy** [comment]
Thanks @aerych!

**Pull request merged**